libtool: cannot execute binary file

The travis libtool fix, broke our build on os x.

```
| No GMP framework detected; in tree GMP will be built
| Run make (MAKEFLAGS='LIBTOOL=bash\ libtool') in libraries/integer-gmp/gmp/gmpbuild...
/usr/bin/libtool: /usr/bin/libtool: cannot execute binary file
make[2]: *** [fib_table.lo] Error 126
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
shakeArgsWith   0.002s    0%                           
Function shake  0.027s    0%                           
Database read   0.006s    0%                           
With database   0.000s    0%                           
Running rules  22.833s   99%  =========================
Total          22.868s  100%                           
Error when running Shake build system:
* inplace/bin/hp2ps
* utils/hp2ps/stage0/build/AreaBelow.o
* OracleQ (DependenciesKey ("utils/hp2ps/stage0/build/.dependencies","utils/hp2ps/stage0/build/AreaBelow.o"))
* utils/hp2ps/stage0/build/.dependencies
* utils/hp2ps/stage0/build/AreaBelow.c.deps
* libraries/integer-gmp/include/ghc-gmp.h
user error (Development.Shake.cmd, system command failed
Command: make -C libraries/integer-gmp/gmp/gmpbuild 'MAKEFLAGS='\''LIBTOOL=bash\ libtool'\'''
Exit code: 2
Stderr:
/usr/bin/libtool: /usr/bin/libtool: cannot execute binary file
make[2]: *** [fib_table.lo] Error 126
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
)
```

--- a/src/Rules/Actions.hs
+++ b/src/Rules/Actions.hs
@@ -94,7 +94,11 @@ runConfigure :: FilePath -> [CmdOption] -> [String] -> Action ()
 runConfigure dir opts args = do
     need [dir -/- "configure"]
     putBuild $ "| Run configure in " ++ dir ++ "..."
-    quietly $ cmd Shell (EchoStdout False) [Cwd dir] "bash configure" opts args
+    quietly $ cmd Shell (EchoStdout False) [Cwd dir] "bash configure" opts' args
+    where
+        -- Always configure with bash.
+        -- This also injects /bin/bash into `libtool`, instead of /bin/sh
+        opts' = opts ++ [AddEnv "CONFIG_SHELL" "/bin/bash"]
 
 runMake :: FilePath -> [String] -> Action ()
 runMake dir args = do

--- a/src/Rules/IntegerGmp.hs
+++ b/src/Rules/IntegerGmp.hs
@@ -114,7 +114,7 @@ integerGmpRules = do
             copyFile integerGmpLibraryFakeH integerGmpLibraryH
         else do
             putBuild "| No GMP framework detected; in tree GMP will be built"
-            runMake integerGmpBuild ["MAKEFLAGS='LIBTOOL=bash\\ libtool'"]
+            runMake integerGmpBuild []
 
             copyFile (integerGmpBuild -/- "gmp.h") integerGmpLibraryInTreeH
             copyFile (integerGmpBuild -/- "gmp.h") integerGmpLibraryH

--- a/src/Rules/IntegerGmp.hs
+++ b/src/Rules/IntegerGmp.hs
@@ -114,7 +114,7 @@ integerGmpRules = do
             copyFile integerGmpLibraryFakeH integerGmpLibraryH
         else do
             putBuild "| No GMP framework detected; in tree GMP will be built"
-            runMake integerGmpBuild []
+            runMake integerGmpBuild ["MAKEFLAGS="]
 
             copyFile (integerGmpBuild -/- "gmp.h") integerGmpLibraryInTreeH
             copyFile (integerGmpBuild -/- "gmp.h") integerGmpLibraryH

